python/cuml/ensemble/randomforestregressor.pyx

```diff
@@ -619,6 +622,14 @@ class RandomForestRegressor(BaseRandomForestModel, RegressorMixin):
             input_to_cuml_array(preds, convert_to_dtype=dtype)
         preds_ptr = preds_m.ptr
 
+        # shortcut for default accuracy metric of r^2
+        if self.accuracy_metric == "r2":
+            stats = r2_score(y_m, preds, handle=self.handle)
+            self.handle.sync()
+            del(y_m)
```

Comment:
These deletes are copied from below. Is there a leak if we don't manually delete these variables?

Reply:
in a distant past there were issues (I'm talking version 0.2 or something very old), but these days we probably are pretty safe without them